WooCommerce/Classes/ViewRelated/CardPresentPayments/CardReaderConnectionController.swift

```diff
@@ -422,13 +427,22 @@ private extension CardReaderConnectionController {
         let cancel = softwareUpdateCancelable.map { cancelable in
             return { [weak self] in
                 self?.state = .cancel
-                let analyticsProperties = [SoftwareUpdateTypeProperty.name: SoftwareUpdateTypeProperty.required.rawValue]
-                ServiceLocator.analytics.track(.cardReaderSoftwareUpdateCancelTapped, withProperties: analyticsProperties)
+                ServiceLocator.analytics.track(
+                    event: WooAnalyticsEvent.InPersonPayments.cardReaderSoftwareUpdateCancelTapped(
+                        forGatewayID: self?.gatewayID,
```

Comment:
For consistency with other places in this file, perhaps we should `guard let self = self` in this closure too

Reply:
Fixed in https://github.com/woocommerce/woocommerce-ios/pull/6123/commits/4db640393735ac24675ea736f4f1280b8febe8c9